src/nupic/utils/Random.hpp

```diff
@@ -30,6 +30,11 @@
 #include <vector>
 #include <utility>
 
+#if defined(NTA_OS_WINDOWS)
```

Comment:
You could undefine directives in CMakeLists.txt using `remove_definitions` in Windows condition. This way you avoid these and future changes in code.